packages/cloud_firestore/cloud_firestore_web/lib/src/interop/firestore_interop.dart

```diff
@@ -160,6 +163,34 @@ external QueryConstraintJsImpl orderBy(
   String? direction,
 ]);
 
+@JS()
+external MemoryLocalCache memoryLocalCache(
+  MemoryCacheSettings? settings,
+);
+
+@JS()
+external MemoryLruGarbageCollector memoryLruGarbageCollector(
+  num? cacheSizeBytes,
+);
+
+@JS()
+external MemoryEagerGarbageCollector memoryEagerGarbageCollector();
+
+@JS()
+external PersistentLocalCache persistentLocalCache(
+  PersistentCacheSettings settings,
+);
+
+@JS()
+external PersistentSingleTabManager persistentSingleTabManager(
+  PersistentSingleTabManagerSettings? settings,
+);
+
+@JS()
+external PersistentMultipleTabManager persistentMultipleTabManager(
+  PersistentSingleTabManagerSettings? settings,
+);
```

Comment:
Is the aim to allow these in the future? (along with `memoryEagerGarbageCollector()` & `memoryLruGarbageCollector()`)

Reply:
Yes we probably need to refactor the front facing API to expose those APIs. I have to draft the API Review